lib/ansible/module_utils/network/meraki/meraki.py

```diff
@@ -236,7 +250,9 @@ def request(self, path, method=None, payload=None):
         self.status = info['status']
 
         if self.status >= 300:
-            self.fail_json(msg='Request failed for {url}: {status} - {msg}'.format(**info))
+            if not payload:
+                self.fail_json(msg='Request failed for {url}: {status} - {msg}'.format(**info))
```

Comment:
I don't see the point here, both errors are identical, so why do this ?
If it fails you wouldn't even know which one actually failed.